pkg/cloud/client.go

```diff
@@ -67,8 +71,11 @@ type SecretConfig struct {
 	StringData Config            `yaml:"stringData"`
 }
 
-var clientCache = map[Config]*client{}
+var clientCache *ttlcache.Cache
 var cacheMutex sync.Mutex
+var log = ctrl.Log.WithName("Client")
```

Comment:
I recall that initializing many different loggers is not ideal - would make sense to instead pass a logger in?

Reply:
Oops I forgot to delete this before committing